src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1074,7 +1073,7 @@ bool entitiesQuery
     connection = getMongoConnection();
     try
     {
-        if ((details == true) && (countP != NULL))
+        if (countP != NULL)
```

Comment:
This part should be executed only when the client has requested details... removing the `details` checking the count is got always, which is inefficient.